presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -5916,6 +5916,9 @@ public void testInformationSchemaFiltering()
         assertQuery(
                 "SELECT table_name FROM information_schema.tables WHERE table_name = 'orders' LIMIT 1",
                 "SELECT 'orders' table_name");
+        assertQuery(
+                "SELECT table_name FROM information_schema.columns WHERE data_type = 'bigint' AND table_name = 'customer' and column_name = 'custkey' LIMIT 1",
```

Comment:
What ATQ subclass i need to run to have this test fail on master?

Also, why "table constraints infinite loop bug"  uses information schema? I mean: information schema is cool and worth being tested, but it's also very special kind of connector.

Reply:
> why "table constraints infinite loop bug" uses information schema?
It wasn't failing with regular tables or at least not with any of the queries I tried out to write the test.

Reply:
I need someone's help to see this fails on current master. Otherwise let's not add the test.

Also, the commit adding regression test should be best squashed with commit fixing the bug.

Reply:
Okay, so the regression was only during work on repurposing the `PickTableLayout` done in this PR. Therefore please squash this with the commit changing `PickTableLayout`